db/memtable_list.cc

```diff
@@ -661,8 +665,15 @@ void MemTableList::Add(MemTable* m, autovector<MemTable*>* to_delete) {
 }
 
 bool MemTableList::TrimHistory(autovector<MemTable*>* to_delete, size_t usage) {
+  // Check if history trim is needed first, so that we can avoid installing a
+  // new MemTableListVersion without installing a SuperVersion (installed based
+  // on return value of this function).
+  if (!current_->HistoryShouldBeTrimmed(usage)) {
```

Comment:
This change relies on the current SuperVersion to always quickly catch up and point to the most recent MemTableListVersion. I did a code audit and found this case where this is not guaranteed, so made this change to make it do the same. This also addressed a comment in the original change: https://github.com/facebook/rocksdb/pull/7296/files#r474976458